helpdesk/models.py

```diff
@@ -517,41 +524,44 @@ def _get_status(self):
         return u'%s%s%s' % (self.get_status_display(), held_msg, dep_msg)
     get_status = property(_get_status)
 
+    def _absolute_uri(self, relative):
+        """
+        Returns an absolute URL for the given relative URL.  This will
+        use self.request, if available (using the same protocol and
+        domain as from the original request), otherwise will make a
+        best guess from the Site or else just a hard-coded domain.
+        """
+        from django.contrib.sites.models import Site
+        # If we have a request object, use it to construct the absolute URL
+        if self.request:
+            return self.request.build_absolute_uri(relative)
+        else:
+            try:
+                site = Site.objects.get_current()
+            except:
+                site = Site(domain='configure-django-sites.com')
+            return u"http://%s%s" % (site.domain, relative)
+
     def _get_ticket_url(self):
         """
         Returns a publicly-viewable URL for this ticket, used when giving
         a URL to the submitter of a ticket.
         """
-        from django.contrib.sites.models import Site
         from django.core.urlresolvers import reverse
-        try:
-            site = Site.objects.get_current()
-        except:
-            site = Site(domain='configure-django-sites.com')
-        return u"http://%s%s?ticket=%s&email=%s" % (
-            site.domain,
+        return self._absolute_uri(u"%s?ticket=%s&email=%s" % (
```

Comment:
I actually considered that. But there are a bunch of cases of '%'-style formatting here.

What we'll probably want to do is to fix all of those in a standalone commit which we can send upstream.

Let's let our usage of this app. cook for a bit, and if we don't decide to throw it away, then we can do a bunch of cleanups targeting upstream.